code/modules/admin/admin_verbs.dm

```diff
@@ -555,6 +555,8 @@ ADMIN_VERB(spawn_debug_full_crew, R_DEBUG, "Spawn Debug Full Crew", "Creates a f
 		if(ishuman(character))
 			GLOB.manifest.inject(character)
 
+		SSmobs.clients_by_zlevel[destination.z] += character
```

Comment:
does not set registered_z so this is broken, also concerned about making the promise of "this is CLIENTED EXPLCITLY". I don't think that's what this verb is for and I don't think it's a promise we can make. Thoughts?

Reply:
```
/mob/living/proc/update_z(new_z) // 1+ to register, null to unregister
	if(registered_z == new_z)
		return
	if(registered_z)
		SSmobs.clients_by_zlevel[registered_z] -= src
	if(isnull(client))
		registered_z = null
		return

	//Check the amount of clients exists on the Z level we're leaving from,
	//this excludes us because at this point we are not registered to any z level.
	var/old_level_new_clients = (registered_z ? SSmobs.clients_by_zlevel[registered_z].len : null)
	//No one is left after we're gone, shut off inactive ones
	if(registered_z && old_level_new_clients == 0)
		for(var/datum/ai_controller/controller as anything in GLOB.ai_controllers_by_zlevel[registered_z])
			controller.set_ai_status(AI_STATUS_OFF)
```

We are setting the mobs client z-Level to the turf they spawn at. I've also tested this and used it to optimize the sound PR by spawning the debug crew with this change applied. The `clients_by_zlevel` list is only used in a few places:

- Weather
- Sound
- AI Controllers

This debug verb also spawns mobs with (empty) minds and jobs.

Reply:
there is no reason this can't change, and I don't expect future users to not rely on these mobs having clients. IDK if it's that big a deal but I'm sort of uncomfortable going THIS far w it. What's the normal usecase for this verb I suppose?

Reply:
> IDK if it's that big a deal but I'm sort of uncomfortable going THIS far w it. 

Understandable. It's just something I had to do to test my code since sound checks for clients.

> What's the normal usecase for this verb I suppose?

Dunno. I've only ever used it like once or twice but it seems like it might be a way to test high pop.